### PR TITLE
#161342828 Configure Authors Haven to use Postgresql as DB Engine

### PR DESCRIPTION
**Pull request by ucheg6**
#### What does this PR do?
- Configure Authors Haven to use Postgresql as DB Engine
#### Description of Task to be completed?
- Refactor app.js file to es6 javascript
- install npm packages necessary to transpile es6 codes to es5
- install Pg pg-hstore and sequelize to configure app to use postgres as database engine
- create database haven-ah with model User and migrated the table

#### How should this be manually tested?
- git clone ` https://github.com/andela/haven-ah-backend.git `
- run npm install to install necessary npm packages
- create dot-env file and setup necessary variables for database configuration
-  run ` createdb haven-ah `to create the database
- run ` sequelize:migrate ` to migrate model
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[#161342828](https://www.pivotaltracker.com/story/show/161342828)
#### Screenshots (if appropriate)
N/A
#### Questions:
N/A

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -0,0 +1,5 @@
+DB_HOST=localhost
```
Comment (seyi-adeleke) [inline .env]: .env should not be in version control

File: app.js [hunk 2]
```diff
@@ -1,93 +1,32 @@
...
 
-const isProduction = process.env.NODE_ENV === "production";
+dotenv.config();
 
-// Create global app object
+// Set up express app
 const app = express();
 
-app.use(cors());
+const port = process.env.PORT || 5000;
```
Comment (seyi-adeleke) [inline app.js]: PORT number should be from the config

File: package.json [hunk 3]
```diff
@@ -4,29 +4,41 @@
...
   "dependencies": {
     "body-parser": "^1.18.3",
     "cors": "^2.8.4",
-    "dotenv": "^6.0.0",
+    "dotenv": "^6.1.0",
     "ejs": "^2.6.1",
     "errorhandler": "^1.5.0",
-    "express": "^4.16.3",
+    "express": "^4.16.4",
     "express-jwt": "^5.3.1",
```
Comment (seyi-adeleke) [inline package.json]: remove unneccsary packages plix

File: server/config/config.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+dotenv.config();
+
+module.exports = {
+  development: {
+    username: process.env.DB_USERNAME,
+    password: process.env.DB_PASSWORD,
+    database: process.env.DB_DATABASE_DEV,
+    host: process.env.DB_HOST,
+    dialect: 'postgres',
+
```
Comment (seyi-adeleke) [inline server/config/config.js]: remove excess whitespaces

File: server/models/User.js [hunk 5]
```diff
@@ -1,57 +1,66 @@
...
+    },
+    role: {
+      type: DataTypes.ENUM('user', 'admin'),
+      allowNull: false,
+    },
+    emailConfirmation: {
+      type: DataTypes.STRING,
+      allowNull: false,
+    },
+    resetToken: {
```
Comment (seyi-adeleke) [inline server/models/User.js]: The resetToken should be nullable

**Review by seyi-adeleke — CHANGES_REQUESTED** [review]

**Comment by seyi-adeleke** [comment]
Also you should be merging into develop

File: server/models/User.js (hunk 5 above)
Comment (ucheg6) [reply server/models/User.js]: I have implemented this

File: app.js [hunk 6]
```diff
@@ -1,93 +1,33 @@
...
 
-const isProduction = process.env.NODE_ENV === "production";
+dotenv.config();
 
-// Create global app object
+// Set up express app
 const app = express();
 
-app.use(cors());
+const { port } = config.development;
```
Comment (seyi-adeleke) [inline app.js]: How's this going to work on production? 🤔

File: package.json [hunk 7]
```diff
@@ -4,33 +4,34 @@
   "description": "A Social platform for the creative at heart",
   "main": "index.js",
   "scripts": {
+    "start-dev": "nodemon --exec babel-node --presets env app.js",
+    "test": "echo \"Error: no test specified\" && exit 1"
```
Comment (seyi-adeleke) [inline package.json]: please fix this 😢

File: package.json [hunk 8]
```diff
@@ -4,33 +4,34 @@
...
+    "pg-hstore": "^2.3.2",
+    "sequelize": "^4.41.0",
+    "slug": "^0.9.1"
   },
   "devDependencies": {
+    "babel-cli": "^6.26.0",
+    "babel-core": "^6.26.3",
+    "babel-preset-env": "^1.7.0",
+    "babel-preset-stage-2": "^6.24.1",
+    "babel-register": "^6.26.0",
```
Comment (seyi-adeleke) [inline package.json]: Here as well. Shouldn't your linter be catching these?

File: server/config/config.js [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+    host: process.env.DB_HOST,
+    port: 5000,
+    dialect: 'postgres',   
+  },
+  test: {
+    username: process.env.DB_USERNAME,
+    password: process.env.DB_PASSWORD,
+    database: process.env.DB_TEST,
+    host: process.env.DB_HOST,
+    port: process.env.DB_PORT,
```
Comment (seyi-adeleke) [inline server/config/config.js]: This is no longer a DB_PORT

File: server/models/User.js [hunk 10]
```diff
@@ -1,57 +1,70 @@
...
+      allowNull: true,
+    },
+    google: {
+      type: DataTypes.STRING,
+      allowNull: true,
+    },
+    twitter: {
+      type: DataTypes.STRING,
+      allowNull: true,
+    },
```
Comment (seyi-adeleke) [inline server/models/User.js]: remove this excess space

**Pull request merged**